File: app/src/main/java/com/felix/madeclass/adapter/TvAdapter.java
package com.felix.madeclass.adapter;

import android.content.Context;
import android.content.Intent;
import android.support.annotation.NonNull;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.LinearLayout;
import android.widget.TextView;

import com.felix.madeclass.DetailTvShowActivity;
import com.felix.madeclass.R;
import com.felix.madeclass.model.TvShow;

import java.util.ArrayList;

public class TvAdapter extends RecyclerView.Adapter<TvAdapter.ViewHolder> {

    private Context context;
    private ArrayList<TvShow> listTv;

    public TvAdapter(Context context, ArrayList<TvShow> listTv) {
        this.context = context;
        this.listTv = listTv;
    }

    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup viewGroup, int i) {
        View v = LayoutInflater.from(context).inflate(R.layout.item_tv, viewGroup, false);

        return new ViewHolder(v);
    }

    @Override
    public void onBindViewHolder(@NonNull final ViewHolder viewHolder, int i) {
        viewHolder.bind(listTv.get(i));

        viewHolder.parentLayout.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                TvShow tvShow = new TvShow();
                tvShow.setTitle(listTv.get(viewHolder.getAdapterPosition()).getTitle());
                tvShow.setOverview(listTv.get(viewHolder.getAdapterPosition()).getOverview());
                tvShow.setRating(listTv.get(viewHolder.getAdapterPosition()).getRating());
                tvShow.setRelease_date(listTv.get(viewHolder.getAdapterPosition()).getRelease_date());
                tvShow.setPhoto(listTv.get(viewHolder.getAdapterPosition()).getPhoto());
                Intent toDetail = new Intent(v.getContext(), DetailTvShowActivity.class);
                toDetail.putExtra("extra_tv", tvShow);
                v.getContext().startActivity(toDetail);
            }
        });

    }

    @Override
    public int getItemCount() {
        return listTv.size();
    }

    public class ViewHolder extends RecyclerView.ViewHolder {

        private TextView txtTitle, txtRating, txtReleaseDate;
        private ImageView imgPoster;
        private LinearLayout parentLayout;

        public ViewHolder(@NonNull View itemView) {
            super(itemView);
            txtRating = itemView.findViewById(R.id.tvTvRating);
            txtReleaseDate = itemView.findViewById(R.id.tvTvReleaseDate);
            txtTitle = itemView.findViewById(R.id.tvTvTitle);
            imgPoster = itemView.findViewById(R.id.ivTvPoster);
            parentLayout = itemView.findViewById(R.id.lvParentTv);
        }

        public void bind(TvShow tvShow){
           txtTitle.setText(tvShow.getTitle());
           txtReleaseDate.setText(tvShow.getRelease_date());
           txtRating.setText(tvShow.getRating());
           imgPoster.setImageResource(tvShow.getPhoto());
        }
    }
}

File: app/src/main/java/com/felix/madeclass/adapter/TvAdapter.kt
package com.felix.madeclass.adapter

import android.content.Context
import android.content.Intent
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.LinearLayout
import android.widget.TextView
import com.bumptech.glide.Glide
import com.felix.madeclass.DetailTvShowActivity
import com.felix.madeclass.R
import com.felix.madeclass.model.TvShow
import java.util.*

class TvAdapter(private val context: Context) : androidx.recyclerview.widget.RecyclerView.Adapter<TvAdapter.ViewHolder>() {


    private val listTv = ArrayList<TvShow>()

    fun setData(tv: ArrayList<TvShow>) {
        listTv.clear()
        listTv.addAll(tv)
        notifyDataSetChanged()
    }

    override fun onCreateViewHolder(viewGroup: ViewGroup, i: Int): ViewHolder {
        val v = LayoutInflater.from(context).inflate(R.layout.item_tv, viewGroup, false)

        return ViewHolder(v)
    }

    override fun onBindViewHolder(viewHolder: ViewHolder, i: Int) {
        viewHolder.bind(listTv[i])

        viewHolder.parentLayout.setOnClickListener { v ->
            val tvShow = TvShow()

            tvShow.title = listTv[viewHolder.adapterPosition].title
            tvShow.overview = listTv[viewHolder.adapterPosition].overview
            tvShow.rating = listTv[viewHolder.adapterPosition].rating
            tvShow.releaseDate = listTv[viewHolder.adapterPosition].releaseDate
            tvShow.photoHigh = listTv[viewHolder.adapterPosition].photoHigh
            tvShow.photoLow = listTv[viewHolder.adapterPosition].photoLow
            tvShow.photoBackdropHigh = listTv[viewHolder.adapterPosition].photoBackdropHigh
            tvShow.photoBackdropLow = listTv[viewHolder.adapterPosition].photoBackdropLow

            val toDetail = Intent(v.context, DetailTvShowActivity::class.java)
            toDetail.putExtra("extra_tv", tvShow)
            v.context.startActivity(toDetail)
        }

    }

    override fun getItemCount(): Int {
        return listTv.size
    }

    inner class ViewHolder(itemView: View) : androidx.recyclerview.widget.RecyclerView.ViewHolder(itemView) {

        private val txtTitle: TextView = itemView.findViewById(R.id.tvTvTitle)
        private val txtRating: TextView = itemView.findViewById(R.id.tvTvRating)
        private val txtReleaseDate: TextView = itemView.findViewById(R.id.tvTvReleaseDate)
        private val imgPoster: ImageView = itemView.findViewById(R.id.ivTvPoster)
        internal val parentLayout: LinearLayout = itemView.findViewById(R.id.lvParentTv)

        fun bind(tvShow: TvShow) {
            txtTitle.text = tvShow.title
            txtReleaseDate.text = tvShow.releaseDate
            txtRating.text = tvShow.rating
            Glide.with(context)
                    .load(tvShow.photoHigh)
                    .fallback(R.drawable.no_image_available)
                    .thumbnail(
                            Glide.with(context)
                                    .load(tvShow.photoLow)
                    )
                    .into(imgPoster)
        }
    }
}
